File: VB/WcfService2/Service1.svc.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports DevExpress.Xpo.DB.Helpers


Namespace WcfService1
	Public Class Service1
		Inherits CachedDataStoreService
		Public Shared MainDataStore As ICachedDataStore
		Shared Sub New()
			Dim connectionString As String = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB")
			Dim dataStore As MSSqlConnectionProvider = CType(XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema), MSSqlConnectionProvider)
			Dim objectsToDispose() As IDisposable
			MainDataStore = CType(MSSql2005SqlDependencyCacheRoot.CreateSqlDependencyCacheRoot(dataStore, New DataCacheConfiguration(DataCacheConfigurationCaching.InList, "Customer"), objectsToDispose), ICachedDataStore)
		End Sub
		Public Sub New()
			MyBase.New(MainDataStore)
		End Sub
	End Class
End Namespace

File: CS/WcfService2/Service1.svc.cs
using System;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using DevExpress.Xpo.DB.Helpers;


namespace WcfService1 {

    public class Service1 : CachedDataStoreService {

        public static ICachedDataStore MainDataStore;

        static Service1() {
            string connectionString = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB");
            MSSqlConnectionProvider dataStore = (MSSqlConnectionProvider)XpoDefault.GetConnectionProvider(
                        connectionString,
                        AutoCreateOption.DatabaseAndSchema
                        );

            IDisposable[] objectsToDispose;
            MainDataStore = (ICachedDataStore)MSSql2005SqlDependencyCacheRoot.CreateSqlDependencyCacheRoot(
                       dataStore,
                       new DataCacheConfiguration(DataCacheConfigurationCaching.InList, "Customer"),
                       out objectsToDispose
                       );
        }

        public Service1() : base(MainDataStore) { }

    }

}
